#NOMES:

# Avaliação


In [ ]:
!pip install numpy  #instala a biblioteca Numpy

ERROR: Invalid requirement: '#instala'

[notice] A new release of pip is available: 23.1.2 -> 23.2.1
[notice] To update, run: C:\Users\Esley\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\python.exe -m pip install --upgrade pip


: 

#### Considere a seguinte lista:

In [ ]:
lista_A = [2, 29, 8, 98, 10, 3, -4, 9, 10, 98]

#### Usando as funções apresentadas, faça os exercícios de 1 à 4:

1) Imprima a soma do maior e o menor elemento da lista:


In [ ]:
soma = max(lista_A) + min(lista_A)
print(soma)

94


In [ ]:
maior = max(lista_A)
menor = min(lista_A)
soma = maior + menor

print(f"A soma do menor numero '{menor}' com o maior numero '{maior}' é de '{soma}'")

A soma do menor numero '-4' com o maior numero '98' é de '94'


2) Imprima o número de ocorrências do maior elemento da lista:

In [ ]:
print(lista_A.count(maior))

2


In [ ]:
qnt_maior = lista_A.count(maior)
print(f"O numero de ocorrências do maior elemento da lista é {qnt_maior}.")

O numero de ocorrências do maior elemento da lista é 2.


3) Imprima o maior número ímpar da lista utilizando um código para calculá-lo:

In [ ]:
lista_impar = []

for i in lista_A:
    if i % 2 == 1:
        lista_impar.append(i)

print(max(lista_impar))
    

29


4) Imprima o maior elemento da lista elevado ao quadrado:

In [ ]:
print(max(lista_A)*max(lista_A))

9604


In [ ]:
maioraoquadrado = maior*maior
print(maioraoquadrado)


9604


5) Crie uma lista com todos os números pares de 0 até 1000 (incluído).

In [ ]:
lista_0a1000 = []

for i in range(0, 1001):
    if i % 2 == 0:
        lista_0a1000.append(i)

print(lista_0a1000)

[0, 2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48, 50, 52, 54, 56, 58, 60, 62, 64, 66, 68, 70, 72, 74, 76, 78, 80, 82, 84, 86, 88, 90, 92, 94, 96, 98, 100, 102, 104, 106, 108, 110, 112, 114, 116, 118, 120, 122, 124, 126, 128, 130, 132, 134, 136, 138, 140, 142, 144, 146, 148, 150, 152, 154, 156, 158, 160, 162, 164, 166, 168, 170, 172, 174, 176, 178, 180, 182, 184, 186, 188, 190, 192, 194, 196, 198, 200, 202, 204, 206, 208, 210, 212, 214, 216, 218, 220, 222, 224, 226, 228, 230, 232, 234, 236, 238, 240, 242, 244, 246, 248, 250, 252, 254, 256, 258, 260, 262, 264, 266, 268, 270, 272, 274, 276, 278, 280, 282, 284, 286, 288, 290, 292, 294, 296, 298, 300, 302, 304, 306, 308, 310, 312, 314, 316, 318, 320, 322, 324, 326, 328, 330, 332, 334, 336, 338, 340, 342, 344, 346, 348, 350, 352, 354, 356, 358, 360, 362, 364, 366, 368, 370, 372, 374, 376, 378, 380, 382, 384, 386, 388, 390, 392, 394, 396, 398, 400, 402, 404, 406, 408, 410, 412, 414, 416, 418, 420,

6) Crie uma lista com todos os números ímpares de 0 até 199 (incluído)

In [ ]:
lista_0a199 = []

for i in range(0, 200):
    if i % 2 == 1:
        lista_0a199.append(i)

print(lista_0a199)

[1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37, 39, 41, 43, 45, 47, 49, 51, 53, 55, 57, 59, 61, 63, 65, 67, 69, 71, 73, 75, 77, 79, 81, 83, 85, 87, 89, 91, 93, 95, 97, 99, 101, 103, 105, 107, 109, 111, 113, 115, 117, 119, 121, 123, 125, 127, 129, 131, 133, 135, 137, 139, 141, 143, 145, 147, 149, 151, 153, 155, 157, 159, 161, 163, 165, 167, 169, 171, 173, 175, 177, 179, 181, 183, 185, 187, 189, 191, 193, 195, 197, 199]


7) Imprima o valor da média de todos os elementos da lista_A:

In [ ]:
print(sum(lista_A) / len(lista_A))

26.3


In [ ]:
valor_medio = soma_lista = sum(lista_A) / len(lista_A)

print(valor_medio)


26.3


8) Crie 3 tuplas para armazenar as notas dos alunos em uma disciplina. Cada tupla deverá conter o nome do aluno (escolhido por você) seguido de suas duas notas.

In [ ]:
Aluno1 = ("Ésley", 9.5, 3.0)
Aluno2 = ("Nathan", 5.3, 7.0)
Aluno3= ("Gomes", 10.0, 9.0)

9) Imprima a soma das notas de cada aluno do exercício 8, use as variáveis criadas por você.
A resposta deve ter o seguinte formato:

O somatório das notas do {Aluno_1} é {nota do Aluno_1}

O somatório das notas do {Aluno_2} é {nota do Aluno_2}

O somatório das notas do {Aluno_3} é {nota do Aluno_3}

In [ ]:
print(f"O somatório das notas do {Aluno1[0]} é {Aluno1[1] + Aluno1[2]}")
print(f"O somatório das notas do {Aluno2[0]} é {Aluno2[1] + Aluno2[2]}")
print(f"O somatório das notas do {Aluno3[0]} é {Aluno3[1] + Aluno3[2]}")

O somatório das notas do Ésley é 12.5
O somatório das notas do Nathan é 12.3
O somatório das notas do Gomes é 19.0


In [ ]:
notas1 = Aluno1[1] + Aluno1[2]
notas2 = Aluno2[1] + Aluno2[2]
notas3 = Aluno3[1] + Aluno3[2]

nome1 = Aluno1[0]
nome2 = Aluno2[0]
nome3 = Aluno3[0]    

print(f"O somatório das notas do {nome1} é {notas1}")
print(f"O somatório das notas do {nome2} é {notas2}")
print(f"O somatório das notas do {nome3} é {notas3}")

O somatório das notas do Ésley é 12.5
O somatório das notas do Nathan é 12.3
O somatório das notas do Gomes é 19.0


10) Imprima o nome e a maior nota de cada aluno do exercício 8, use as variáveis criadas por você.

A maior nota do {Aluno_1} é {maior_nota_do_Aluno_1} e a sua menor nota é {menor_nota_do_Aluno_1}

A maior nota do {Aluno_2} é {maior_nota_do_Aluno_2} e a sua menor nota é {menor_nota_do_Aluno_2}

A maior nota do {Aluno_3} é {maior_nota_do_Aluno_3} e a sua menor nota é {menor_nota_do_Aluno_3}

In [ ]:
notas_aluno1 = [Aluno1[1], Aluno1[2]]
notas_aluno2 = [Aluno2[1], Aluno2[2]]
notas_aluno3 = [Aluno3[1], Aluno3[2]]

print(f"A maior nota do {Aluno1[0]} é {max(notas_aluno1)} e a sua menor nota é {min(notas_aluno1)}")
print(f"A maior nota do {Aluno2[0]} é {max(notas_aluno2)} e a sua menor nota é {min(notas_aluno2)}")
print(f"A maior nota do {Aluno3[0]} é {max(notas_aluno3)} e a sua menor nota é {min(notas_aluno3)}")

A maior nota do Ésley é 9.5 e a sua menor nota é 3.0
A maior nota do Nathan é 7.0 e a sua menor nota é 5.3
A maior nota do Gomes é 10.0 e a sua menor nota é 9.0


In [ ]:
nota_aluno1_1b = Aluno1[1]
nota_aluno1_2b = Aluno1[2]
notas1 = Aluno1[1] + Aluno1[2]
if nota_aluno1_1b > nota_aluno1_2b:
    maior_aluno1 = nota_aluno1_1b
    menor_aluno1 = nota_aluno1_2b
else:
    maior_aluno1 = nota_aluno1_2b    
    menor_aluno1 = nota_aluno1_1b

print(f"A maior nota do {nome1} é {maior_aluno1} e a sua menor nota é {menor_aluno1}")

A maior nota do Ésley é 9.5 e a sua menor nota é 3.0


11) Construa um dicionário para armazenar os CNPJs de 5 empresas e o lucro ou prejuízo acumulado no ano de 2022. A chave deve ser o nome da empresa e o valor deve ser o cnpj.








* Petrobras
CNPJ 33.000.167/0661-29
faturamento 2021: R$ 452,7bilhões

* JBS
CNPJ 02.916.265/0001-60
faturamento 2021: R$ 350,7bilhões

* Vale
CNPJ 33.592.510/0001-54
faturamento 2021: R$ 293,5bilhões

* Raízen
CNPJ 08.070.508/0001-78
faturamento 2021: R$ 137,8bilhões

* Vibra
CNPJ 34.274.233/0001-02
faturamento 2021: R$ 130,1bilhões

In [ ]:
cnpj = {
    "Petrobras": ["33.000.167/0661-29", "152,70"],
    "JBS": ["02.916.265/0001-60", "350,70"],
    "Vale": ["33.592.510/0001-54", "293,50"],
    "Raízen": ["08.070.508/0001-78", "137,80"],
    "Vibra": ["34.274.233/0001-02", "130,10"],
}

print(type(cnpj))
print(cnpj["Petrobras"])

<class 'dict'>
['33.000.167/0661-29', '152,70']


12) Usando o dicionário criado no exercício 11, imprima o valor do menor e do maior faturamento e a média do faturamento das 5 empresas. A reposta deve ter o seguinte formato:

* A empresa com o maior faturamento {valor do faturamento} é {nome da empresa}.
* A empresa com o menor faturamento {valor do faturamento} é {nome da empresa}.
* A média do faturamento das cinco empresas em 2021 foi de {média do faturamento}.

In [ ]:
for i in cnpj.values(): 
  valor = i[1]
  print(type(valor))

  if valor > aux:
    aux = valor
    print(aux)

<class 'str'>


NameError: name 'aux' is not defined

In [ ]:
lista_cnpj = list(cnpj.items())

print(lista_cnpj)

lista_valores = []

for i in lista_cnpj:    
    lista_valores.append(i[1][1])


print (lista_valores)
# valor_medio = str(sum(lista_valores)) / str(len(lista_valores))

print(f"A empresa com o maior faturamento {max(lista_valores)} é "),
print(f"A empresa com o menor faturamento {min(lista_valores)} é ")
# print(F"A média do faturamento das cinco empresas em 2021 foi de {valor_medio}.")

13) Sabendo que A = {0, 1, 2, 3, 4, 5, 6}, B = {6, 7, 8, 9} e C = {2, 4, 6, 8, 10}, quais são os elementos do conjunto (A∩B)UC?

In [ ]:
A = {0, 1, 2, 3, 4, 5, 6}
B = {6, 7, 8, 9}
C = {2, 4, 6, 8, 10}

intersessao = A.intersection(B)
resposta =  intersessao.union(C)

print(resposta)

{2, 4, 6, 8, 10}


14) Os alunos de uma escola devem se inscrever em DUAS MODALIDADES esportivas no início do ano, porém, as inscrições são abertas e alguns alunos podem ter se inscrito em apenas uma modalidade ou em três. As modalidades esportivas são: NATAÇÃO, CICLISMO E CORRIDA. Você é o responsável pelas inscrições e deverá analisar as listagens com as inscrições e realizar as seguintes ações corretivas:

14.1. Para cada uma das três modalidades, verifique se o mesmo aluno se inscreveu mais de uma vez e exclua exclua os nomes duplicados.

14.2. Em seguida, contabilize a quantidade de alunos inscritos em
cada modalidade e imprima os três valores

14.3 Imprima uma listagem com o nome de todos os alunos nas 3 modalidades e a quantidade total de alunos, somadas as 3 modalidades.

14.4 Imprima uma listagem com o nome de cada aluno e as modalidades em que está inscrito

14.5 Imprima uma listagem com os alunos que se inscreveram em apenas 1 ou em 3 modalidades

In [ ]:
natacao = ["Nicolas da Silva", "Marcos Santos", "Luciana Barbosa", "Ana Schimidt", "Marcos Santos", "Joana Ramos", "André Consentino", "Marcela Rangel", "Luciana Barbosa", "Carlos Antunes"]
ciclismo = ["Nicolas da Silva", "Marcos Santos", "Luciana Barbosa", "Ana Schimidt", "Marcos Santos","Marcela Rangel", "Michele Souza", "Marcelo Bach", "Luis Pereira"]
corrida = ["Marcos Santos", "Luciana Barbosa", "Ana Schimidt", "Marcos Santos", "Luis Pereira", "Joana Ramos", "André Consentino","Marcela Rangel", "Luis Gomes", "Luciana Barbosa"]

Foram inscritos 10 na modalidade natação
Foram inscritos 9 na modalidade ciclismo
Foram inscritos 10 na modalidade corrida


In [ ]:
natacao = set(natacao)
ciclismo = set(ciclismo)
corrida = set(corrida)

print(f'Foram inscritos {len(natacao)} na modalidade natação')
print(f'Foram inscritos {len(ciclismo)} na modalidade ciclismo')
print(f'Foram inscritos {len(corrida)} na modalidade corrida')

Foram inscritos 8 na modalidade natação
Foram inscritos 8 na modalidade ciclismo
Foram inscritos 8 na modalidade corrida


In [ ]:
aluno = natacao.intersection(ciclismo)
aluno = aluno.intersection(corrida)

print(f"Os seguintes alunos estão matriculados nas três modalidades {aluno}, total: {len(aluno)}")

Os seguintes alunos estão matriculados nas três modalidades {'Ana Schimidt', 'Marcos Santos', 'Marcela Rangel', 'Luciana Barbosa'}, total: 4


In [ ]:
for i in natacao:
    print(i)

André Consentino
Marcos Santos
Joana Ramos
Luciana Barbosa
Nicolas da Silva
Ana Schimidt
Carlos Antunes
Marcela Rangel


15) Abaixo é apresentado um dicionário contendo os seguintes dados de alguns vencedores do Prêmio Nobel da Paz:
{Ano da premiação: (Nome, País de Origem, Ano de Nascimento, Motivação do prêmio, gênero (Feminino, Masculino ou Organização))}

Faça o que se pede nos subtitens 15.1 e 15.2

In [ ]:
nobel = {2011: [['Ellen Jonhson Sirleaf', 'Liberia', 1938, 'woman rights', "F"],['Leymah Gbowee', 'Liberia', 1972, 'woman rights', 'F'], ['Tawakkol', 'Liberia', 1979, 'woman rights', 'F' ]],
         2012: ['Eurpean Union', 'Europe', 1952, 'human rights','O'],
         2013: ['Organization for the Prohibition of Chemical Weapons', 'The Netherlands', 1997, 'eliminate chemical weapons', 'O'],
         2014: [['Kailash Satyarthi', 'India', 1954, 'children education', 'M'], ['Malala Yousafzai', 'Pakistan', 1997, 'children education', 'O']],
         2015: ['National Dialogue Quartet', 'Tunisia', 2011, 'democracy', 'O'],
         2016: ['Juan Manuel Santos', 'Colombia', 1951, 'end of civil war', 'O'],
         2017: ['International Campaign to Abolish Nuclear Weapons', 'Australia', 2007, 'nuclear weapons prohibition', 'O'],
         2018: ['Denis Mukwege', 'Congo', 1955, 'end sexual violation at war', 'M'],
         2019: ['Abiy Ahmed Ali', 'Ethiopia', 1976, 'international conflicts', 'M'],
         2020: ['World Food Programme', 'Italy', 1961, 'combat hunger', 'O']}

15.1) Quais foram os vencedores do Prêmio Nobel da Paz em 2011?

In [ ]:
for chave, valor in nobel.items():
    if chave == 2011:
        for i in valor:
            print(i[0])

Ellen Jonhson Sirleaf
Leymah Gbowee
Tawakkol


15.2) Imprima a relação dos países de origem dos laureados com o Prêmio Nobel da Paz na última década.

In [ ]:
for i in nobel.values():
    print(type(i[1]))
    
    # if len(i) == <class 'list'>:        
    #     for i2 in i:
    #         print(i)
    #         # print(i2[1])

<class 'list'>
<class 'str'>
<class 'str'>
<class 'list'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>
<class 'str'>


16) Em uma competição de salto em distância cada atleta tem direito a três saltos. O resultado do atleta será determinado pela média dos três valores saltos.

Você deve fazer um programa que receba:

* o nome e as três distâncias alcançadas pelo atleta em seus saltos (via input())
* informe o nome, os saltos e a média dos saltos.
* O programa deve ser encerrado quando não for informado o nome do atleta.
* A saída do programa deve ser conforme o exemplo abaixo:

Atleta: Rodrigo

Primeiro Salto: 3 m

Segundo Salto: 2 m

Terceiro Salto: 4 m

Resultado final:

Atleta: Rodrigo

Saltos: 3 - 2 - 4

Média dos saltos: 3 m


17) Qual é a diferença entre o **while** e **for**?

RESPOSTA: While é usado para quando você quer fazer uma repetição que não tenha um limite estabelecido. Por exemplo, quando você quer ficar gerando numeros aleatorios e apenas quando o numero for 948 o while vai ser interrompido. Já a função for serve para quando você quer fazer uma repetição que você já sabe o limite dela.

18) Faça um Programa que peça a idade e a altura de 3 pessoas, armazene cada informação na sua respectiva lista e imprima a idade e a altura de cada pessoa.

In [29]:
i = 0
pessoa = {}

while i < 3:    
    print(i)
    nome = input('Digite o nome:')
    altura = input('Digite a altura:')

    pessoa[i] = {"nome": nome, "altura": altura}

    i = i + 1

<class 'dict'>
0
1
2


In [30]:
for i in pessoa:
    print(f'A {i}º pessoa cadadastra chama-se {pessoa[i]["nome"]}, e mede {pessoa[i]["altura"]} de altura.')

A 0º pessoa cadadastra chama-se Ésley, e mede 190 de altura.
A 1º pessoa cadadastra chama-se Nathan, e mede 180 de altura.
A 2º pessoa cadadastra chama-se Gomes, e mede 170 de altura.


19) Dados os dois vetores abaixo (listas), faça um programa que leia os dois vetores e gere um terceiro vetor de 20 elementos, cujos valores deverão ser compostos pelos elementos intercalados dos dois outros vetores.

In [34]:
a = [0,2,4,6,8,10,12,14,16,18]
b = [1,3,5,7,9,11,13,15,17,19]

intercalada = []

for a,b in zip(a, b):
        intercalada.append(a)
        intercalada.append(b)
print(intercalada)

[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19]


20) Faça UMA FUNÇÃO que recebe um valor n e que imprima:
  
    1
  
    2   2
  
    3   3   3
  
    .....
  
    n   n   n   n   n   n  ... n

21) Faça uma FUNÇÃO que recebe um valor n e que imprima:
  
  1
  
  1   2
  
  1   2   3
  
  .....
  
  1   2   3   ...  n

22) Faça uma FUNÇÃO que retorne o reverso de um número inteiro passado por parâmetro. Por exemplo: 127 -> 721.

In [39]:
inteiro = input("Digite um numero inteiro")

print(inteiro[::-1])


001


23) Faça uma função que receba dois vetores com 5 elementos aleatórios e inteiros cada. Sua função deverá retornar um terceiro vetor de 20 elementos, cujos valores deverão ser compostos pelos elementos concatenados dos dois outros vetores.

24) Faça uma função que receba uma string e retorne a quantidade de consoantes que existem nela.

In [68]:
palavra = input("Digite uma palavra")

lista_consoantes = []

for i in palavra:    
    if i == 'a':
        lista_consoantes.append(i)
    if i == 'e':
        lista_consoantes.append(i)
    if i == 'i':
        lista_consoantes.append(i)
    if i == 'o':
        lista_consoantes.append(i)
    if i == 'u':
        lista_consoantes.append(i)

print(f"A quantidade de consoantes da palavra '{palavra}' é de {len(lista_consoantes)}.")
      

A quantidade de consoantes da palavra 'desenho' tem 3


25) Altere o exercício 24 para que o retorno seja um dicionário, onde a chaves (keys) são as consoantes e os valores (values) são a quantidade de vezes em que a consoante aparece na string